### PR TITLE
[Fix] Fix CascadeRoIHead export when reg_class_agnostic=True in box_head 

### PR DESCRIPTION
**Pull request by irexyc**
## Motivation

Try to fix https://github.com/open-mmlab/mmdeploy/issues/1896

### CODE REVIEW

File: mmdeploy/codebase/mmdet/models/roi_heads/cascade_roi_head.py [hunk 1]
```diff
@@ -59,24 +59,21 @@ def cascade_roi_head__simple_test(ctx, self, x, proposals, img_metas,
...
+            rois = self.bbox_head[i].regress_by_class(rois, bbox_label,
+                                                      bbox_pred, img_metas[0])
 
     cls_score = sum(ms_scores) / float(len(ms_scores))
-    bbox_pred = bbox_pred.reshape(batch_size, num_proposals_per_img, 4)
-    rois = rois.reshape(batch_size, num_proposals_per_img, -1)
-    scale_factor = img_metas[0].get('scale_factor', None)
+    cls_score = cls_score.unsqueeze(0)
+    rois = rois.unsqueeze(0)
+    bbox_pred = bbox_pred.unsqueeze(0)
```
Comment (AllentDan) [inline mmdeploy/codebase/mmdet/models/roi_heads/cascade_roi_head.py:75]: Does it mean that `batch_size` must be 1?

File: mmdeploy/codebase/mmdet/models/roi_heads/cascade_roi_head.py [hunk 2]
```diff
@@ -59,24 +59,21 @@ def cascade_roi_head__simple_test(ctx, self, x, proposals, img_metas,
...
+
         ms_scores.append(cls_score)
         if i < self.num_stages - 1:
-            assert self.bbox_head[i].reg_class_agnostic
-            new_rois = self.bbox_head[i].bbox_coder.decode(
-                rois[..., 1:], bbox_pred, max_shape=max_shape)
-            rois = new_rois.reshape(-1, new_rois.shape[-1])
-            # Add dummy batch index
-            rois = torch.cat([rois.new_zeros(rois.shape[0], 1), rois], dim=-1)
+            if self.bbox_head[i].custom_activation:
```
Comment (RunningLeon) [inline mmdeploy/codebase/mmdet/models/roi_heads/cascade_roi_head.py:65]: Since convnext is supported, should we add convnext to regression yaml?

**Review by RunningLeon — APPROVED** [review]
LGTM

**Comment by Rinatum** [comment]
Good afternoon, the same problem on the ``main`` branch (mmdeploy-1x). Could you apply same solution to the new mmdeploy versions?

**Comment by cly234** [comment]
Seems like the main branch did not update this, how can i solve this? By manually cherry-picking the changes of the commit to my local packages of mmdeploy?

**Pull request merged**